server/src/main/java/org/opensearch/cluster/action/index/NodeMappingRefreshAction.java

```diff
@@ -80,11 +92,70 @@ public NodeMappingRefreshAction(TransportService transportService, MetadataMappi
     }
 
     public void nodeMappingRefresh(final DiscoveryNode clusterManagerNode, final NodeMappingRefreshRequest request) {
+        new NodeMappingRefreshClusterManagerAction(clusterManagerNode, request).run();
+    }
+
+    private void sendNodeMappingRefreshToClusterManager(
+        final DiscoveryNode clusterManagerNode,
+        final NodeMappingRefreshRequest request,
+        ActionListener listener
+    ) {
         if (clusterManagerNode == null) {
             logger.warn("can't send mapping refresh for [{}], no cluster-manager known.", request.index());
             return;
         }
-        transportService.sendRequest(clusterManagerNode, ACTION_NAME, request, EmptyTransportResponseHandler.INSTANCE_SAME);
+        transportService.sendRequest(clusterManagerNode, ACTION_NAME, request, new EmptyTransportResponseHandler(ThreadPool.Names.SAME) {
+            @Override
+            public void handleException(TransportException exp) {
+                listener.onFailure(exp);
+            }
+        });
+    }
+
+    /**
+     * RetryableAction for performing retires for cluster manager throttling.
+     */
+    private class NodeMappingRefreshClusterManagerAction extends RetryableAction {
+
+        private final DiscoveryNode clusterManagerNode;
+        private final NodeMappingRefreshRequest request;
+        private static final int BASE_DELAY_MILLIS = 10;
+        private static final int MAX_DELAY_MILLIS = 10;
+
+        private NodeMappingRefreshClusterManagerAction(DiscoveryNode clusterManagerNode, NodeMappingRefreshRequest request) {
+            super(
+                logger,
+                threadPool,
+                TimeValue.timeValueMillis(BASE_DELAY_MILLIS),
+                TimeValue.timeValueMillis(Integer.MAX_VALUE), // Shard tasks are internal and don't have timeout
+                new ActionListener() {
+                    @Override
+                    public void onResponse(Object o) {}
+
+                    @Override
+                    public void onFailure(Exception e) {
+                        logger.warn("Mapping refresh for [{}] failed due to [{}]", request.index, e.getMessage());
+                    }
+                },
+                BackoffPolicy.exponentialEqualJitterBackoff(BASE_DELAY_MILLIS, MAX_DELAY_MILLIS),
+                ThreadPool.Names.SAME
+            );
+            this.clusterManagerNode = clusterManagerNode;
+            this.request = request;
+        }
+
+        @Override
+        public void tryAction(ActionListener listener) {
+            sendNodeMappingRefreshToClusterManager(clusterManagerNode, request, listener);
+        }
+
```

Comment:
This would make onboarding more actions to task throttling framework more tedious. Can we abstract out all complexities and avoid creating a new class per action altogether.

Reply:
Ideally new actions which needs to be perform on master should extend `TransportClusterManagerNodeAction`. All this retry logic is abstracted out in it. 

These two actions refresh-mapping/shard-state was not extending this `TransportClusterManagerNodeAction` and directly sending request to master node, so we need to add retryable logic here as well. 

**[NOTE]:** To keep this Data node side PR clean, I am going to remove this changes of NodeMappingRefreshAction and ShardStateAction. Will add those in PR where we onboard those task in these framework.